src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspaceService.scala

```diff
@@ -76,4 +97,58 @@ class WorkspaceService(protected val argUserInfo: UserInfo, val rawlsDAO: RawlsD
     }
   }
 
+  def exportWorkspaceAttributesTSV(workspaceNamespace: String, workspaceName: String, filename: String): Future[PerRequestMessage] = {
+    rawlsDAO.getWorkspace(workspaceNamespace, workspaceName) map { workspaceResponse =>
+      val attributes = workspaceResponse.workspace.get.attributes.filterKeys(_ != AttributeName("default", "description"))
+      val headerString = "workspace:" + (attributes map { case (attName, attValue) => attName.name }).mkString("\t")
+      val valueString = (attributes map { case (attName, attValue) => impPlainAttributeFormat.write(attValue) }).mkString("\t")
+      RequestCompleteWithHeaders((StatusCodes.OK, headerString + "\n" + valueString),
+        HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> filename)),
+        HttpHeaders.`Content-Type`(`text/plain`))
+    }
+  }
+
+  def importAttributesFromTSV(workspaceNamespace: String, workspaceName: String, tsvString: String): Future[PerRequestMessage] = {
+    Try(TSVParser.parse(tsvString)) match {
+      case Failure(regret) => Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, regret.getMessage))
```

Comment:
this try/failure case is equivalent to `withTSVFile` in EntityClient. can you make them both use the same code?